pelican/plugins/seo/seo.py

```diff
@@ -43,28 +36,67 @@ def plugin_initializer(settings):
     logger.info("SEO plugin initialized")
 
 
+def get_plugin_settings(context):
+    """Get settings in the pelican config file from the given context
+
+    .. note:: Pelican's settings take precedence over the plugin's default settings.
+
+    SEO report settings:
+
+        - SEO_REPORT
+        - SEO_ARTICLES_LIMIT
+        - SEO_PAGES_LIMIT
+
+    SEO enhancer settings:
+
+        - SEO_ENHANCER
+        - SEO_ENHANCER_OPEN_GRAPH
+        - SEO_ENHANCER_TWITTER_CARDS
+
+    :return: Dictionary of settings for the plugin
+    :rtype: <dict>
+    """
+
+    def get_seo_variables(data):
+        """Get variables starting with 'SEO_' & values in the given dictionary
+        :rtype: <dict>
+        """
+        return {
+            var_name: value
+            for var_name, value in data.items()
+            if var_name.startswith("SEO_")
+        }
+
+    plugin_settings = get_seo_variables(vars(settings))
+    plugin_settings.update(get_seo_variables(context))
+    return plugin_settings
+
+
 def run_seo_report(generators):
     """Run SEO report creation if SEO_REPORT is enabled in settings."""
 
+    plugin_settings = get_plugin_settings(generators[0].context)
```

Comment:
Can you check first if `generators` is not None? I prefer to avoid errors like `NoneType object has no attribute context` in case of Pelican doesn't return a generator.

Reply:
I thought of it but in practice it can not happen, there is at least the generator of articles and pages in this iterable. It is part of the heart of Pelican, and can not be removed (without really wanting). I think if they are not present, the user will have other problems to manage :D What do you think?

Reply:
I totally agree with you. It's just that I prefer to have custom errors, instead of Python errors, to avoid users opening issues on this repo while it's a Pelican issue. But I'm ok to keep this like it.